src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1038,25 +1069,23 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             // since that method calls EventQueue.TryComplete. Without
                             // TryComplete, we may miss diagnostics.
                             var hostDiagnostics = analyzerDriver.GetDiagnosticsAsync(compilation).Result;
-                            diagnostics.AddRange(hostDiagnostics);
 
-                            if (!diagnostics.IsEmptyWithoutResolution)
-                            {
-                                // Apply diagnostic suppressions for analyzer and/or compiler diagnostics from diagnostic suppressors.
-                                analyzerDriver.ApplyProgrammaticSuppressions(diagnostics, compilation);
+                            // Apply diagnostic suppressions for analyzer compiler diagnostics from diagnostic suppressors.
+                            hostDiagnostics = analyzerDriver.ApplyProgrammaticSuppressions(hostDiagnostics, compilation);
```

Comment:
>hostDiagnostics = analyzerDriver.ApplyProgrammaticSuppressions(hostDiagnostics, compilation); [](start = 28, length = 93)

Are we making an assumption that any diagnostics reported after the previous ```ApplyProgrammaticSuppressions``` call added above cannot be elevated or suppressed? Can we assert that in some way so that a violation of the invariant is caught early? Alternatively, consider collecting diagnostics after that ```ApplyProgrammaticSuppressions``` call into a separate bag and combining it with ```hostDiagnostics``` for the purpose of this ```ApplyProgrammaticSuppressions``` call, then merging the result into ```diagnostics```.

Reply:
I have taken the latter approach, but implemented the fix in the analyzer driver to avoid each caller of `ApplyProgrammaticSuppressions` having to track processed diagnostics.